backend/main.py

```
@@ -178,8 +312,64 @@ def post_product(product: schemas.ProductCreate, db: Session = Depends(get_db),
+@app.get("/get-product/{product_id}", response_model=schemas.ProductOut)
+def get_product(product_id: int,db: Session = Depends(get_db)):
+    product = db.query(models.Product).filter(models.Product.product_id == product_id).first()
+
+    if not product:
+        raise HTTPException(status_code=404, detail="Product not found.")
+
+    return product
+
+
+
+@app.get("/get-all-products", response_model=List[schemas.ProductOut])
```

Ruff (0.8.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)